schemas/constructs/core.json

```
@@ -0,0 +1,39 @@
+{
+  "$id": "https://schemas.meshery.io/core.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "description": "Reusable core schema elements",
+  "definitions": {
+    "inputString": {
+      "type": "string",
+      "pattern": "^[a-zA-Z_][a-zA-Z0-9_]*$"
```

```suggestion
      "pattern": "^[a-zA-Z_][a-zA-Z0-9_-]*[a-zA-Z0-9_]$"
```

This allow hyphens within the string (but not at the beginning or end).

Should this be "inputString" or just "string"?